src/wrapper/jnienv.rs

```diff
@@ -248,7 +248,14 @@ impl<'a> JNIEnv<'a> {
         S: Into<JNIString>,
         T: Desc<'a, JClass<'a>>,
     {
-        self.throw((class, msg))
+        let class = class.lookup(self)?;
+        let msg = msg.into();
+        let res: i32 = jni_unchecked!(self.internal, ThrowNew, class.into_inner(), msg.as_ptr());
+        if res < 0 {
```

Comment:
Could you please change that to better match the [spec](https://docs.oracle.com/en/java/javase/11/docs/specs/jni/functions.html#thrownew):

> Returns 0 on success; a negative value on failure.

```
if res == 0 → Ok
else → Err(…)
```

Reply:
Sure, no problem. I took that from `throw`, should I update `throw` as well? Should I panic on a positive result or also simply return `Err`?

Reply:
> should I update throw as well

Would be wonderful!

> Should I panic on a positive result or also simply return Err?

I think it's fine to simply return `Err` as we can expect the JNI interface functions to follow the specification.